packages/google_adsense/lib/src/ad_unit_widget_web.dart

```diff
@@ -0,0 +1,201 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:developer';
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+
+import 'package:flutter/widgets.dart';
+import 'package:web/web.dart' as web;
+
+import '../google_adsense.dart';
+import 'js_interop/adsbygoogle.dart';
+
+/// Widget displaying an ad unit
+class AdUnitWidgetWeb extends AdUnitWidget {
+  // TODO(sokoloff06): consider builder?
+  /// Constructs [AdUnitWidgetWeb]
+  AdUnitWidgetWeb({
+    required String adClient,
+    required String adSlot,
+    required bool isAdTest,
+    required Map<String, dynamic> additionalParams,
+    String? cssText,
+    super.key,
+  })  : _adClient = adClient,
+        _adSlot = adSlot,
+        _isAdTest = isAdTest,
+        _additionalParams = additionalParams {
+    _insElement
+      ..className = 'adsbygoogle'
+      ..style.display = 'block';
+    if (cssText != null && cssText.isNotEmpty) {
+      _insElement.style.cssText = cssText;
+    }
+    final Map<String, String> dataAttrs =
+        Map<String, String>.of(<String, String>{
+      AdUnitParams.AD_CLIENT: 'ca-pub-$_adClient',
+      'adSlot': _adSlot,
+    });
+    if (_isAdTest) {
+      dataAttrs.addAll(<String, String>{_AD_TEST_KEY: 'on'});
+    }
+    for (final String key in dataAttrs.keys) {
+      _insElement.dataset
+          .setProperty(key as JSString, dataAttrs[key]! as JSString);
```

Comment:
Casting `as JSString` is (maybe?) going to break when we use this in WASM, try using `.toJS` instead from the new JS-interop:

```dart
  .setProperty(key.toJS, dataAttrs[key]!.toJS);
```

(See the docs: **JavaScript interop > JS Types > [Conversions](https://dart.dev/interop/js-interop/js-types#conversions)**)

Reply:
I think I got rid of all as JS* casts, will test WASM compatibility as well

Reply:
Compiles and runs with WASM succesfully! See adsense-test.visokolov.com